packages/server/src/requestPipeline.ts

```diff
@@ -75,6 +76,10 @@ function isBadUserInputGraphQLError(error: GraphQLError): boolean {
   );
 }
 
+function isUnknownOperationNameError(error: GraphQLError): boolean {
```

Comment:
We also need to catch the case where multiple operations are provided but no operation name is given: https://github.com/graphql/graphql-js/blob/ffe37cb7f501e65e445c30764a1b34f659562756/src/execution/execute.ts#L275

(I don't think we need to handle the [third error](https://github.com/graphql/graphql-js/blob/ffe37cb7f501e65e445c30764a1b34f659562756/src/execution/execute.ts#L296) in that function because IIUC this happens if there are no operations in the document, but I think that's a validation error: the parser requires there to be *some* top-level definition, and ExecutableDefinitionsRule requires all definitions to be operations or fragments. And NoUnusedFragmentsRule requires all fragments to be used (transitively) by an operation.)

That probably means that the name UnknownOperationNameError should change. Maybe to OperationResolveError or OperationResolutionError or something (going with our `didResolveOperation` hook).

Then um.  We could just decide that if `requestContext.operation` is falsey, we turn any error we get out of `execute` into an OperationResolutionError, assuming our logic matches that in graphql-js... which I think is probably true.

BTW, if we do want to stick with error string matching, it raises the question of whether the error is precisely the same when `execute` is gateway or a custom executor... It looks like `buildOperationContext` in gateway happens to use the same error strings but that's a strong assumption...

(Of course hopefully one day graphql-js will just let us use its operation-resolving code directly, cc @IvanGoncharov.)